packages/@aws-cdk/cdk/lib/core/construct.ts

```diff
@@ -299,6 +304,22 @@ export class Construct {
     return ret;
   }
 
+  /**
+   * TODO docs
+   */
+  public apply(aspect: IAspect): void {
+    this.aspects.push(aspect);
+  }
+
+  /**
+   * TODO docs
+   * can this be private?
+   */
+  public visit(): void {
```

Comment:
agree - even in Wikipedia they call apply `Accept` and `Visit` does the work. I'm open to name changes.

Reply:
My naming suggestion here is semantic meaning from the perspective of the user, not mechanic meaning from the perspective of the implementor.

For the user, it looks like they are *applying* a tag to a construct, and the word *apply* is nice and generic enough that we can apply it (hah) to other types of aspects as well.

A mechanically correct alternative name would be `attachAspect()`, but it doesn't read as nicely.

Compare:

```ts
// Customer first (what it does)
construct.apply(new cdk.Tag('key', 'value'));

// Implementor first (how it works)
construct.attachAspect(new cdk.TagAspect('key', 'value'));
```

At least, im(ns)ho :)

Reply:
Also, I wouldn't call this particular function `visit()`, I would call this something like `invokeAspects()`.

And yes, it shouldn't be public, that will only serve to confuse people. It's not too bad, it should be invoked from `App.run()` which is itself a construct so it can access `protected` members.